src/ApiHandler.ts

```diff
@@ -139,6 +125,72 @@ export default class ApiHandler {
 			}
 		}
 
+		const perByteFee = api.consts.transactionPayment.transactionByteFee;
+		const extrinsicBaseWeight = api.consts.system.extrinsicBaseWeight;
+		const coefficients = api.consts.transactionPayment.weightToFee[0];
+		const coeffFrac = coefficients.coeff_frac;
+		const coeffInt = coefficients.coeff_integer;
+		
+		// TODO: We assume that the polynomial has one positive term of degree one.
+		// const degree = coefficients.degree;
+		// const negative = coefficients.negative;
+
+		for (let idx = 0; idx < block.extrinsics.length; ++idx) {
+			if (!extrinsics[idx].paysFee) {
+				continue;
+			}
+
+			try {
+				if (api.runtimeVersion.specName.toString() === 'kusama') {
+					extrinsics[idx].info = api.createType('RuntimeDispatchInfo',await api.rpc.payment.queryInfo(block.extrinsics[idx].toHex(), parentHash));
+					continue;
+				}
+
+				const xtEvents = extrinsics[idx].events;
+				const success = 'system.ExtrinsicSuccess';
+				const failure = 'system.ExtrinsicFailed';
```

Comment:
I doubt it's much of a performance change, but maybe move these outside the for loop?

Reply:
I would prefer to keep these inside the loop close to where they are being used (this is an important principle of writing good code). Declaring a variable should not have an appreciable effect on performance in this case. This change was made to address the line length issues that Alexander brought up...normally I would have just defined these inline and not pulled them out into a variable.

Reply:
But you're also using them in [101](https://github.com/paritytech/substrate-api-sidecar/pull/50/files#diff-fdb5fe31b3fb59e594f5a04dabf5c38eR101) and [105](https://github.com/paritytech/substrate-api-sidecar/pull/50/files#diff-fdb5fe31b3fb59e594f5a04dabf5c38eR105). Why not declare them once and use them consistently?

Reply:
Fair enough...I went ahead and changed that. To answer your question, though, in order to keep the diff as small as possible, I was staying focused on the code blocks that I was writing. From my POV, it's better to keep PRs like this focused on the specific feature being implemented or bug being fixed. There should be _separate_ PRs that go back and refactor the existing code. But that's just my $0.02.